Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,143 @@
 
 namespace Smartbox\Integration\FrameworkBundle\Components\Queues;
 
-use Psr\Log\LoggerAwareInterface;
-use Psr\Log\LoggerAwareTrait;
-use Smartbox\Integration\FrameworkBundle\Components\Queues\Handler\AmqpQueueHandler;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\ConsumerInterface;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use PhpAmqpLib\Message\AMQPMessage;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\PhpAmqpLibDriver;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
 use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
 use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
-use Smartbox\Integration\FrameworkBundle\Service;
 
-class AsyncQueueConsumer extends Service implements ConsumerInterface, LoggerAwareInterface
+/**
+ * Class PhpAmqpSignalConsumer.
+ */
+class AsyncQueueConsumer extends AbstractAsyncConsumer
 {
-    use IsStopableConsumer;
     use UsesSmartesbHelper;
     use UsesSerializer;
-    use LoggerAwareTrait;
     use UsesExceptionHandlerTrait;
 
     /**
-     * @var QueueManager
+     * Consumer identifier name.
      */
-    private $manager;
+    const CONSUMER_TAG = 'amqp-consumer-%s-%s';
 
     /**
      * @var string
      */
-    private $format;
+    private $format = QueueDriverInterface::FORMAT_JSON;
 
     /**
-     * AsyncQueueConsumer constructor.
+     * @var PhpAmqpLibDriver
+     */
+    private $driver;
+
+    /**
+     * Set the driver to this class with the properties fulfilled.
      *
-     * @param QueueManager $manager
-     * @param string       $format
+     * @param $driver
      */
-    public function __construct(QueueManager $manager, string $format = 'json')
+    public function setDriver($driver)
```

Comment:
When setting up the driver, we should force that the driver follows a specific interface, in this case **AsyncQueueDriverInterface** to make sure the methods of the driver called in this class exists

Reply:
Ok. Suggestion applied.